packages/page-council/src/Motions/Voting.tsx

```diff
@@ -42,6 +53,18 @@ function Voting ({ hash, idNumber, isDisabled, members, prime, proposal }: Props
 
   const isPrime = prime?.toString() === accountId;
 
+  const voteExtrinsic = api.tx.council.vote(hash, idNumber, voteValue);
+  const closeExtrinsic = api.tx.council.close.meta.args.length === 4
+    ? api.tx.council.close(hash, idNumber, proposalWeight, proposalLength)
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore older version (2 params)
+    : api.tx.council.close(hash, idNumber);
+
+  // vote and close if this vote ends the vote
+  const extrinsic = willPass || willFail
+    ? api.tx.utility.batch([voteExtrinsic, closeExtrinsic])
+    : voteExtrinsic;
```

Comment:
We don't want to create these each render - this is quite expensive.

Reply:
Right, I'll use `useMemo`